client/rpc/debug/src/lib.rs

```diff
@@ -359,13 +362,35 @@ where
 			.map_err(|e| internal_err(format!("Fail to read blockchain db: {:?}", e)))?
 			.unwrap_or_default();
 
+		// Get DebugRuntimeApi version
+		let trace_api_version = if let Ok(Some(api_version)) =
+			api.api_version::<dyn DebugRuntimeApi<B>>(parent_block_hash)
+		{
+			api_version
+		} else {
+			return Err(internal_err(
+				"Runtime api version call failed (trace)".to_string(),
+			));
+		};
+
 		// Trace the block.
 		let f = || -> RpcResult<_> {
-			api.initialize_block(parent_block_hash, &header)
-				.map_err(|e| internal_err(format!("Runtime api access error: {:?}", e)))?;
+			let result = if trace_api_version >= 5 {
+				// The block is initialized inside "trace_block"
+				api.trace_block(parent_block_hash, exts, eth_tx_hashes, &header)
+			} else {
+				// Pre pallet-message-queue
+
+				// Initialize block: calls the "on_initialize" hook on every pallet
+				// in AllPalletsWithSystem
+				api.initialize_block(parent_block_hash, &header)
```

Comment:
It might worth adding an additional brief comment elaborating a bit more about the reason for these changes (just a suggestion). What do you think?

Reply:
Done